src/ring_theory/finiteness.lean

```diff
@@ -260,10 +274,50 @@ end
 /-- If `f : A →ₐ[R] B` is surjective with finitely generated kernel and `A` is finitely presented,
 then so is `B`. -/
 lemma of_surjective {f : A →ₐ[R] B} (hf : function.surjective f) (hker : f.to_ring_hom.ker.fg)
-  (hfp : finitely_presented R A) : finitely_presented R B :=
+  (hfp : finite_presentation R A) : finite_presentation R B :=
 equiv (quotient hker hfp) (ideal.quotient_ker_alg_equiv_of_surjective hf)
 
-end finitely_presented
+/-- An algebra is finitely presented if and only if it is a quotient of a polynomial ring whose
+variables are indexed by a fintype by a finitely generated ideal. -/
+lemma iff_quotient_mv_polynomial' : finite_presentation R A ↔ ∃ (ι : Type u_2) [fintype ι]
+  (f : (_root_.mv_polynomial ι R) →ₐ[R] A), (surjective f) ∧ f.to_ring_hom.ker.fg :=
+begin
+  split,
+  { rintro ⟨n, ⟨f, hf⟩⟩,
+    refine ⟨ulift (fin n), by apply_instance, f.comp (mv_polynomial.rename equiv.ulift),
```

Comment:
You'll probably have an easier time with:
```suggestion
    refine ⟨ulift (fin n), by apply_instance, f.comp (mv_polynomial.alg_equiv_of_equiv R equiv.ulift).to_alg_hom,
```
since then the surjectivity proof is just unfolding coes and saying "alg_equivs are surjective"

Reply:
Thank you!